lib/Input.js

```diff
@@ -37,7 +47,9 @@ class Input extends React.Component {
   }
 
   componentDidUpdate (prevProps) {
-    if (this.props.autoresize && prevProps.query !== this.props.query) {
+    if (prevProps.autoresize !== this.props.autoresize ||
+      (this.props.autoresize && RESIZE_PROPS.some((prop) => this.props[prop] !== prevProps[prop]))
+    ) {
```

Comment:
My thinking was that the check would be slower than a few unnecessary recalcs with the overall benefit of simply having less logic to maintain.

I'm happy to ship this as-is TBH, apologies for being slow - I'm super busy ATM